File: 003: Largest Prime/LargestPrimeFactor.java
import java.util.ArrayList;

public class LargestPrimeFactor {
    public static void main(String[] args) {
        factors(13195);

    }
    /*
    The prime factors of 13195 are 5, 7, 13 and 29.
    
    What is the largest prime factor of the given number?
     */

    public static void factors(int number) {
        int largestPrime = 0;
        ArrayList<Integer> factors = new ArrayList<>();


        for (int i = 2; i < number; i++) {
            if (number % i == 0) {
                factors.add(i);
            }
        }
        for (int i = 0; i < factors.size(); i++){
            if (isPrime(factors.get(i)) && factors.get(i) > largestPrime) {
                largestPrime = factors.get(i);
            }

        }
        System.out.println(largestPrime);
    }

    public static boolean isPrime(int factor) {
        boolean prime = true;
        int i = 2;
        while (i < factor) {
            if (factor % i == 0) {
                prime = false;
                break;
            }
            i++;
        }
        return prime;
    }
}

File: 003: Largest Prime/LargestPrimeFactor.py
def main():
    num = int(input("Enter the number to get its largest prime factor: "))
    print(largest_factor(num))


def largest_factor(number):
    largest_prime = 0
    factors = []
    prime_factors = []

    for i in range(2, number):
        if number % i == 0:
            factors.append(i)

    for num in factors:
        if is_prime(num) == True and num > largest_prime:
            prime_factors.append(num)
    # print(prime_factors)
    return max(prime_factors)


def is_prime(factor):
    prime = True
    for i in range(2, factor):
        if factor % i == 0:
            prime = False
            break
        i += 1

    return prime


main()
